### PR TITLE
Add view transformations migration guide

### PR DESCRIPTION
**Pull request by IceSentry**
Closes https://github.com/bevyengine/bevy-website/issues/825

### CODE REVIEW

**Review by rparrett — APPROVED** [review]
Oh wow, was that the only breaking change there? Seems like it.

**Comment by IceSentry** [comment]
Yeah, it was mostly just adding new functions that we had to manually write before. I missed the fact that it now uses vec3 instead of vec4 but I updated it now.

**Comment by rparrett** [comment]
> Yeah, it was mostly just adding new functions that we had to manually write before. I missed the fact that it now uses vec3 instead of vec4 but I updated it now.

I actually just hit that right now while completing the migration I was working on, and ended up with the same thing. Cool.

File: content/learn/migration-guides/0.11-0.12/_index.md [hunk 1]
```diff
@@ -1531,3 +1531,23 @@ App::new()
...
+// 0.11
+#import bevy_pbr::mesh_functions::mesh_position_world_to_clip
+fn mesh_position_local_to_clip(model: mat4x4<f32>, vertex_position: vec4<f32>) -> vec4<f32> {
+    let world_position = mesh_position_local_to_world(model, vertex_position);
+    return mesh_position_world_to_clip(world_position);
+}
+
+// 0.12
+#import bevy_pbr::view_transformations::position_world_to_clip;
+fn mesh_position_local_to_clip(model: mat4x4<f32>, vertex_position: vec4<f32>) -> vec4<f32> {
```
Comment (DGriffin91) [inline content/learn/migration-guides/0.11-0.12/_index.md:1549]: This is incorrect. There is no `mesh_position_local_to_clip` function in `bevy_pbr::view_transformations`. None of the functions in `bevy_pbr::view_transformations` take a mat as a parameter.

Comment (IceSentry) [reply content/learn/migration-guides/0.11-0.12/_index.md:1549]: This is showing the migration of the `mesh_position_local_to_clip` function that uses `position_world_to_clip`. This isn't in `view_transformations` it was to show the import.

Comment (IceSentry) [reply content/learn/migration-guides/0.11-0.12/_index.md:1549]: it was copied from `mesh_functions.wgsl`

Comment (rparrett) [reply content/learn/migration-guides/0.11-0.12/_index.md:1549]: Yeah okay, checks out.

I wonder if the function signature is just a bit of noise we don't need here though.

Comment (DGriffin91) [reply content/learn/migration-guides/0.11-0.12/_index.md:1549]: I see what you mean now. I only very briefly looked at this. The similarity in names between everything here may also make things confusing in the guide. idk though

Comment (TrialDragon) [reply content/learn/migration-guides/0.11-0.12/_index.md:1549]: I agree that the function signature is confusing and noise.

Comment (rparrett) [reply content/learn/migration-guides/0.11-0.12/_index.md:1549]: Maybe just show in diff format like this:
```diff
-#import bevy_pbr::mesh_functions::mesh_position_world_to_clip
+#import bevy_pbr::view_transformations::position_world_to_clip;
 fn mesh_position_local_to_clip(model: mat4x4<f32>, vertex_position: vec4<f32>) -> vec4<f32> {
     let world_position = mesh_position_local_to_world(model, vertex_position);
-    return mesh_position_world_to_clip(world_position);
+    return position_world_to_clip(world_position.xyz);
 }
 ```

**Review by alice-i-cecile — APPROVED** [review]
This is more useful than nothing. Merging now: feel free to clean up further in a follow-up.

**Comment by alice-i-cecile** [comment]
Ugh, nasty merge conflicts... This needs to target the newly renamed file.

**Pull request closed without merging**